### ISSUE TITLE
Transformations translate resized image to the SouthEast

### ISSUE THREAD

**Issue by beaudet** [ruven/iipsrv#98]
It seems this code in Transforms.cc

  float xscale = (float)(width-1) / (float)resampled_width;
  float yscale = (float)(height-1) / (float)resampled_height;

should be

  float xscale = (float)(width) / (float)resampled_width;
  float yscale = (float)(height) / (float)resampled_height;

otherwise resized images (when compared to an image magick resize using the original source) are shifted slightly SouthEast.

e.g.

ImageMagick
![image](https://cloud.githubusercontent.com/assets/77699/24420296/672ef6f2-13c0-11e7-8478-d7fd420fb609.png)

vs

IIP bilinear
![image](https://cloud.githubusercontent.com/assets/77699/24420310/74d47ca0-13c0-11e7-956e-94802484f09d.png)

**Comment by ruven** [comment]
Hmmm, not sure why I wrote it in this way. There may have been a good reason, but I don't know what it could be! The scaling is as you suggest for the nearest neighbour implementation.

In any case, I just pushed a fix in @736d9acf043b6bb5c2c13acc005446e782f0d545

**Comment by beaudet** [comment]
There might be another tweak required to the existing bilinear filter.  With four pixel samples in the as-is bilinear filter, the "center" is essentially the upper left pixel so we're still 0.5 pixels off on average since the sample is actually down and to the right of the true center in the source image.  I think recomputing the "center" by 0.5 pixels would do the trick.

**Comment by beaudet** [comment]
Actually, there is no problem with IIP after making the initial change.  There does, however, seem to be a problem with the vips utility rescaling images and that's the source of the problem.  See screenshots below - all the tiles were generated from the same source image, but notice on the right side how the edge of the circle gets clipped just slightly in lower resolution.  ImageMagick seems to scale the image properly for the same resolution. Hmm...

![image](https://cloud.githubusercontent.com/assets/77699/24574016/7dc503d8-165a-11e7-91cd-decd5a2ba1a7.png)

![image](https://cloud.githubusercontent.com/assets/77699/24574022/95cde9cc-165a-11e7-9ed8-a2493ef387f6.png)

**Comment by beaudet** [comment]
I have confirmed that there is an ongoing resize bug in vips that is reproducible on the latest up-to-date RHEL6 and RHEL7 systems using either vips 7.x or the latest 8.4 version of vips.  See the differences between the images below which were taken from the pyramidal tiff.  The first was created using imagemagick and the second with vips 8.4.5 on RHEL7.   Aside from being slower, we had found a few years back that IM creates intermittent corrupted ptifs that IIP can't open.  I think the only option is to find the bug in vips or the library it's using and fix it - or switch to IM if the PTIF generation has been fixed since we last looked at it about 4 or 5 years ago.  Any advice appreciated.

One of the resolutions from an ImageMagick generated pyramidal tiff
![image](https://cloud.githubusercontent.com/assets/77699/26126104/627bc2e4-3a52-11e7-8fa5-8ec515175035.png)

and same resolution from vips generated pyramidal tiff
![image](https://cloud.githubusercontent.com/assets/77699/26126132/8053bcfe-3a52-11e7-9a12-8f86e7164faf.png)
